prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -120,11 +144,15 @@ void efa_rdm_cq_proc_ibv_recv_rdma_with_imm_completion(
 
 	efa_cntr_report_rx_completion(&ep->base_ep.util_ep, flags);
 
-	/* Recv with immediate will consume a pkt_entry, but the pkt is not
-	   filled, so free the pkt_entry and record we have one less posted
-	   packet now. */
-	ep->efa_rx_pkts_posted--;
-	efa_rdm_pke_release_rx(pkt_entry);
+	if (!efa_rdm_cq_wc_is_unsolicited(ibv_cq_ex)) {
```

Comment:
`efa_rdm_cq_proc_ibv_recv_rdma_with_imm_completion` currently needs pkt_entry as an input to get the ep where the completion is targeting. That requires we always have a valid wr_id for the wc. Are we sure it will be the case? @mrgolin